atomicfile/atomicfile.go

```diff
@@ -10,17 +10,30 @@ package atomicfile // import "tailscale.com/atomicfile"
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 )
 
 // WriteFile writes data to filename+some suffix, then renames it
 // into filename.
 func WriteFile(filename string, data []byte, perm os.FileMode) error {
 	tmpname := filename + ".new.tmp"
```

Comment:
ioutil.TempFile instead.

Reply:
Can you guarantee the temp file will be on the same filesystem as the original?

Reply:
The first argument to TempFile is the dir to put it in: https://golang.org/pkg/io/ioutil/#TempFile